// ng-animate/esm5/lib/rotate.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { animation, style, animate, keyframes } from '@angular/animations';
import { DEFAULT_TIMING } from './utils';
/**
 * @param {?} origin
 * @param {?} degrees
 * @return {?}
 */
export function rotateInDirection(origin, degrees) {
    return animation(animate('{{ timing }}s {{ delay }}s', keyframes([
        style({
            'transform-origin': '{{ origin }}',
            opacity: 0,
            transform: 'rotate3d(0, 0, 1, {{ degrees }})',
            offset: 0
        }),
        style({
            'transform-origin': '{{ origin }}',
            opacity: 1,
            transform: 'none',
            offset: 1
        })
    ])), {
        params: { timing: DEFAULT_TIMING, delay: 0, origin: origin, degrees: degrees }
    });
}
export var /** @type {?} */ rotateIn = rotateInDirection('center', '-200deg');
export var /** @type {?} */ rotateInDownLeft = rotateInDirection('left bottom', '-45deg');
export var /** @type {?} */ rotateInDownRight = rotateInDirection('right bottom', '45deg');
export var /** @type {?} */ rotateInUpLeft = rotateInDirection('left bottom', '45deg');
export var /** @type {?} */ rotateInUpRight = rotateInDirection('right bottom', '-90deg');
/**
 * @param {?} origin
 * @param {?} degrees
 * @return {?}
 */
export function rotateOutDirection(origin, degrees) {
    return animation(animate('{{ timing }}s {{ delay }}s', keyframes([
        style({
            'transform-origin': '{{ origin }}',
            opacity: 1,
            transform: 'none',
            offset: 0
        }),
        style({
            'transform-origin': '{{ origin }}',
            opacity: 0,
            transform: 'rotate3d(0, 0, 1, {{ degrees }})',
            offset: 1
        })
    ])), {
        params: { timing: DEFAULT_TIMING, delay: 0, origin: origin, degrees: degrees }
    });
}
export var /** @type {?} */ rotateOut = rotateOutDirection('center', '200deg');
export var /** @type {?} */ rotateOutDownLeft = rotateOutDirection('left bottom', '45deg');
export var /** @type {?} */ rotateOutDownRight = rotateOutDirection('right bottom', '-45deg');
export var /** @type {?} */ rotateOutUpLeft = rotateOutDirection('left bottom', '-45deg');
export var /** @type {?} */ rotateOutUpRight = rotateOutDirection('right bottom', '90deg');
