// ngx-collapsible/ngx-collapsible.es5.js
import { Directive, Input, NgModule, ViewContainerRef } from '@angular/core';
import { CommonModule } from '@angular/common';
var CollapsibleDirective = (function () {
    /**
     * @param {?} viewContainer
     */
    function CollapsibleDirective(viewContainer) {
        var _this = this;
        this.viewContainer = viewContainer;
        this.selected = [];
        this.openHeaderClass = 'header-open';
        this.openContentClass = 'content-open';
        this.collapsedHeaderClass = 'header-collapsed';
        this.collapsedContentClass = 'content-collapsed';
        this.headers = [];
        this.contents = [];
        this.toggleAccordion = function (key) {
            if (_this.selected[0] === key) {
                _this.selected = [];
            }
            else {
                _this.selected = [key];
            }
        };
        this.toggleClasses = function (element, openClass, collapsedClass, isCollapsed) {
            if (isCollapsed === void 0) { isCollapsed = true; }
            var /** @type {?} */ allClasses = element.className.split(' ');
            var /** @type {?} */ classToAdd = isCollapsed ? collapsedClass : openClass;
            var /** @type {?} */ addIndex = allClasses.indexOf(classToAdd);
            if (addIndex === -1) {
                allClasses.push(classToAdd);
            }
            var /** @type {?} */ classToRemove = isCollapsed ? openClass : collapsedClass;
            var /** @type {?} */ removeIndex = allClasses.indexOf(classToRemove);
            if (removeIndex > -1) {
                allClasses.splice(removeIndex, 1);
            }
            element.className = allClasses.join(' ');
        };
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    CollapsibleDirective.prototype.ngOnChanges = function (changes) {
        if (this.accordion) {
            this.selected = this.selected.slice(0, 1);
        }
        this.updateView();
    };
    /**
     * @return {?}
     */
    CollapsibleDirective.prototype.ngAfterViewInit = function () {
        var _this = this;
        var /** @type {?} */ element = this.viewContainer.element.nativeElement;
        this.headers = [].slice.call(element.querySelectorAll('[collapsible-header]'));
        this.contents = [].slice.call(element.querySelectorAll('[collapsible-content]'));
        this.headers.forEach(function (el) {
            el.addEventListener('click', function () {
                var /** @type {?} */ key = el.getAttribute('collapsible-header');
                if (key === null) {
                    return;
                }
                if (_this.accordion) {
                    _this.toggleAccordion(key);
                }
                else {
                    _this.toggleCollapsible(key);
                }
                _this.updateView();
            });
        });
        this.updateView();
    };
    /**
     * @param {?} key
     * @return {?}
     */
    CollapsibleDirective.prototype.toggleCollapsible = function (key) {
        var /** @type {?} */ index = this.selected.indexOf(key);
        if (index === -1) {
            this.selected.push(key);
        }
        else {
            this.selected.splice(index, 1);
        }
    };
    /**
     * @return {?}
     */
    CollapsibleDirective.prototype.updateView = function () {
        var _this = this;
        this.headers.forEach(function (el) {
            var /** @type {?} */ key = el.getAttribute('collapsible-header');
            var /** @type {?} */ collapse = _this.selected.indexOf(key) === -1;
            _this.toggleClasses(el, _this.openHeaderClass, _this.collapsedHeaderClass, collapse);
        });
        this.contents.forEach(function (el) {
            var /** @type {?} */ key = el.getAttribute('collapsible-content');
            var /** @type {?} */ collapse = _this.selected.indexOf(key) === -1;
            _this.toggleClasses(el, _this.openContentClass, _this.collapsedContentClass, collapse);
        });
    };
    return CollapsibleDirective;
}());
CollapsibleDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngxCollapsible]'
            },] },
];
/**
 * @nocollapse
 */
CollapsibleDirective.ctorParameters = function () { return [
    { type: ViewContainerRef, },
]; };
CollapsibleDirective.propDecorators = {
    'accordion': [{ type: Input },],
    'selected': [{ type: Input },],
    'openHeaderClass': [{ type: Input },],
    'openContentClass': [{ type: Input },],
    'collapsedHeaderClass': [{ type: Input },],
    'collapsedContentClass': [{ type: Input },],
};
var CollapsibleModule = (function () {
    function CollapsibleModule() {
    }
    return CollapsibleModule;
}());
CollapsibleModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [CollapsibleDirective],
                exports: [CollapsibleDirective],
            },] },
];
/**
 * @nocollapse
 */
CollapsibleModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { CollapsibleModule, CollapsibleDirective as ɵa };
